File: walkthrough/wwwroot/bridge/test/integration/navigationJourney.js
Bridge.assembly("WalkthroughClientScripts", function ($asm, globals) {
    "use strict";

    Bridge.define("WalkthroughClientScripts.Tests.NavigationJourney", {
        $metadata : function () { return {"att":1048577,"a":2,"m":[{"a":2,"isSynthetic":true,"n":".ctor","t":1,"sn":"ctor"},{"a":2,"n":"Script","is":true,"t":8,"sn":"Script","rt":System.Void}]}; }
    });
});

sap.ui.require(System.Array.init(["sap/ui/test/opaQunit", "sap/ui/demo/walkthrough/test/integration/pages/app"], System.String), function () {
    QUnit.module("Navigation");

    opaTest("Should open the hello dialog", function (Given, When, Then) {
        // Arrangements
        Given.iStartMyAppInAFrame((sap.ui.require.toUrl("sap/ui/demo/walkthrough/test/mockserver") || "") + ".html");

        // Actions
        When['WalkthroughClientScripts.Tests.OnTheAppPage'].iPressTheSayHelloWithDialogButton();

        // Assertions
        Then['WalkthroughClientScripts.Tests.OnTheAppPage'].iShouldSeeTheHelloDialog();

        // Cleanup
        Then['WalkthroughClientScripts.Tests.OnTheAppPage'].iTeardownMyAppFrame();
    });
});

File: WalkthroughClientScripts/test/integration/NavigationJourney.cs
using Bridge;
using System;
using System.Linq;
using static Retyped.qunit;
using UI5;
using UI5.Tests;

namespace WalkthroughClientScripts.Tests
{
    [FileName("test/integration/navigationJourney.js")]
    public class NavigationJourney
    {
        [Init(InitPosition.Bottom)]
        public static void Script()
        {
            sap.ui.require(new string[] {
                    "sap/ui/test/opaQunit",
                    "sap/ui/demo/walkthrough/test/integration/pages/app"
                },
                new Action(() => {
                    QUnit.module("Navigation");

                    TestsGlobals.opaTest("Should open the hello dialog", 
                        (sap.ui.test.Opa5 Given, Map<sap.ui.test.Opa5> When, Map<sap.ui.test.Opa5> Then) => {
                            // Arrangements
                            Given.iStartMyAppInAFrame(sap.ui.require_toUrl("sap/ui/demo/walkthrough/test/mockserver") + ".html");
                            
                            // Actions
                            When.Get<OnTheAppPage>().iPressTheSayHelloWithDialogButton();

                            // Assertions
                            Then.Get<OnTheAppPage>().iShouldSeeTheHelloDialog();

                            // Cleanup
                            Then.Get<OnTheAppPage>().iTeardownMyAppFrame();
                        }
                    );
                })
            );
        }
    }
}
